packages/network-controller/src/NetworkController.ts

```diff
@@ -991,201 +1293,627 @@ export class NetworkController extends BaseController<
+        this.state.networkConfigurationsByChainId,
+      )) {
+        const rpcEndpoint = networkConfiguration.rpcEndpoints.find(
+          (existingRpcEndpoint) =>
+            URI.equal(rpcEndpointFields.url, existingRpcEndpoint.url),
+        );
+        if (rpcEndpoint) {
+          throw new Error(
+            `Cannot add network that points to same RPC endpoint as existing network for chain ${networkConfiguration.chainId} ('${networkConfiguration.name}')`,
+          );
+        }
       }
     }
-    if (!ticker) {
+
+    if (
+      [...new Set(setOfRpcEndpointFields)].length <
+      setOfRpcEndpointFields.length
+    ) {
       throw new Error(
-        'A ticker is required to add or update networkConfiguration',
+        'Cannot add network: Each entry in rpcEndpoints must be unique',
       );
     }
 
-    const autoManagedNetworkClientRegistry =
-      this.#ensureAutoManagedNetworkClientRegistryPopulated();
+    if ([...new Set(networkClientIds)].length < networkClientIds.length) {
+      throw new Error(
+        'Cannot add network: Each entry in rpcEndpoints must have a unique networkClientId',
+      );
+    }
 
-    const existingNetworkConfiguration = Object.values(
-      this.state.networkConfigurations,
-    ).find(
-      (networkConfig) =>
-        networkConfig.rpcUrl.toLowerCase() === rpcUrl.toLowerCase(),
+    if (infuraRpcEndpoints.length > 1) {
+      throw new Error(
+        'Cannot add network: There cannot be more than one Infura RPC endpoint',
+      );
+    }
+
+    const infuraRpcEndpoint = setOfRpcEndpointFields.find(
+      (rpcEndpointFields) => rpcEndpointFields.type === RpcEndpointType.Infura,
     );
-    const upsertedNetworkConfigurationId = existingNetworkConfiguration
-      ? existingNetworkConfiguration.id
-      : random();
-    const networkClientId = upsertedNetworkConfigurationId;
-
-    const customNetworkClientRegistry =
-      autoManagedNetworkClientRegistry[NetworkClientType.Custom];
-    const existingAutoManagedNetworkClient =
-      customNetworkClientRegistry[networkClientId];
-    const shouldDestroyExistingNetworkClient =
-      existingAutoManagedNetworkClient &&
-      existingAutoManagedNetworkClient.configuration.chainId !== chainId;
-    if (shouldDestroyExistingNetworkClient) {
-      existingAutoManagedNetworkClient.destroy();
+    if (infuraRpcEndpoint) {
+      const infuraNetworkName = deriveInfuraNetworkNameFromRpcEndpointUrl(
+        infuraRpcEndpoint.url,
+      );
+      const infuraNetworkNickname = NetworkNickname[infuraNetworkName];
+      const infuraChainId = ChainId[infuraNetworkName];
+      if (chainId !== infuraChainId) {
+        throw new Error(
+          // This is a string.
+          // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+          `Cannot add network with chain ID ${chainId} and Infura RPC endpoint for '${infuraNetworkNickname}' which represents ${infuraChainId}, as the two conflict`,
```

Comment:
Looks like this might have been a false alarm. I'm not seeing any error messages here. Every so often I need to restart the typescript server in vscode to get the correct output.

```suggestion
          `Cannot add network with chain ID ${chainId} and Infura RPC endpoint for '${infuraNetworkNickname}' which represents ${infuraChainId}, as the two conflict`,
```

Reply:
Hmm, interesting. I've noticed that having old TypeScript build files around can confuse the linter. I'll try clearing those out and re-running `yarn lint`.

Reply:
No need, I'm seeing the errors now! Looks like there was something about by local config that was off. Sorry for the confusion. 

We probably should investigate why `restrict-template-expressions` raises so many false positives for `any` though.